service/matching/task_queue_partition_manager.go

```diff
@@ -322,15 +325,67 @@ func (pm *taskQueuePartitionManagerImpl) GetUserDataManager() userDataManager {
 
 // GetAllPollerInfo returns all pollers that polled from this taskqueue in last few minutes
 func (pm *taskQueuePartitionManagerImpl) GetAllPollerInfo() []*taskqueuepb.PollerInfo {
-	return pm.defaultQueue.GetAllPollerInfo()
+	ret := pm.defaultQueue.GetAllPollerInfo()
+	pm.versionedQueuesLock.RLock()
+	defer pm.versionedQueuesLock.RUnlock()
+	for _, ptqm := range pm.versionedQueues {
+		info := ptqm.GetAllPollerInfo()
+		ret = append(ret, info...)
+	}
+	return ret
 }
 
-func (pm *taskQueuePartitionManagerImpl) HasPollerAfter(accessTime time.Time) bool {
-	return pm.defaultQueue.HasPollerAfter(accessTime)
+func (pm *taskQueuePartitionManagerImpl) HasAnyPollerAfter(accessTime time.Time) bool {
+	if pm.defaultQueue.HasPollerAfter(accessTime) {
+		return true
+	}
+	pm.versionedQueuesLock.RLock()
+	defer pm.versionedQueuesLock.RUnlock()
+	for _, ptqm := range pm.versionedQueues {
+		if ptqm.HasPollerAfter(accessTime) {
+			return true
+		}
+	}
+	return false
+}
+
+func (pm *taskQueuePartitionManagerImpl) HasPollerAfter(buildId string, accessTime time.Time) bool {
+	if buildId == "" {
+		return pm.defaultQueue.HasPollerAfter(accessTime)
+	}
+	pm.versionedQueuesLock.RLock()
+	vq, ok := pm.versionedQueues[buildId]
+	pm.versionedQueuesLock.RUnlock()
+	if !ok {
+		return false
+	}
+	return vq.HasPollerAfter(accessTime)
 }
 
 func (pm *taskQueuePartitionManagerImpl) DescribeTaskQueue(includeTaskQueueStatus bool) *matchingservice.DescribeTaskQueueResponse {
-	return pm.defaultQueue.DescribeTaskQueue(includeTaskQueueStatus)
+	pm.versionedQueuesLock.RLock()
+	defer pm.versionedQueuesLock.RUnlock()
+	responses := make([]*matchingservice.DescribeTaskQueueResponse, len(pm.versionedQueues)+1)
+	responses[0] = pm.defaultQueue.DescribeTaskQueue(includeTaskQueueStatus)
+	i := 1
+	for _, ptqm := range pm.versionedQueues {
+		responses[i] = ptqm.DescribeTaskQueue(includeTaskQueueStatus)
+		i++
+	}
+
+	mergeResponses := func(responses []*matchingservice.DescribeTaskQueueResponse) *matchingservice.DescribeTaskQueueResponse {
+		pollers := make([]*taskqueuepb.PollerInfo, 0)
+		for _, e := range responses {
+			pollers = append(pollers, e.GetPollers()...)
+		}
+		return &matchingservice.DescribeTaskQueueResponse{
+			Pollers: pollers,
+			// one TaskQueueStatus doesn't make sense for multiple queues, but this response will change with the new API, so leaving it for now
+			TaskQueueStatus: responses[0].GetTaskQueueStatus(),
+		}
+	}
+
+	return mergeResponses(responses)
```

Comment:
why not something like this?
```suggestion
		resp := &matchingservice.DescribeTaskQueueResponse{
			Pollers: pm.GetAllPollersInfo(),
		}
		if includeTaskQueueStatus {
		     resp.TaskQueueStatus = pm.defaultQueue.DescribeTaskQueue(true).TaskQueueStatus
		}
		return resp
```

BTW, I think once we add the DesribeTaskQueuePartition API, the DescribeTaskQueue methods of partition manager and physical queue manager should be renamed to something like LegacyDescribeTaskQueue.

Reply:
Nice, I think it makes sense to include the task queue status of the default queue.

I did the whole thing with "merge responses" instead of just `GetAllPollers` because I thought that I would be using this same function for the new Describe API, and that there would be more info to merge then, so I was going to use the same "merge" logic for that. But you're right, it should be a totally separate function for the legacy vs the new behavior, so your suggestion is better.